apollo-router/src/plugins/telemetry/apollo.rs

```diff
@@ -19,19 +21,26 @@ use super::metrics::apollo::studio::SingleStatsReport;
 use super::tracing::apollo::TracesReport;
 use crate::plugin::serde::deserialize_header_name;
 use crate::plugin::serde::deserialize_vec_header_name;
+use crate::plugins::telemetry::apollo_exporter::proto::ReferencedFieldsForType;
+use crate::plugins::telemetry::apollo_exporter::proto::ReportHeader;
+use crate::plugins::telemetry::apollo_exporter::proto::StatsContext;
+use crate::plugins::telemetry::apollo_exporter::proto::Trace;
 use crate::plugins::telemetry::config::SamplerOption;
-use crate::spaceport::ReferencedFieldsForType;
-use crate::spaceport::ReportHeader;
-use crate::spaceport::StatsContext;
-use crate::spaceport::Trace;
+use crate::plugins::telemetry::tracing::BatchProcessorConfig;
+
+pub(crate) const ENDPOINT_DEFAULT: &str =
+    "https://usage-reporting.api.apollographql.com/api/ingress/traces";
+// Magic invalid URL. There's no good way of checking the validity of defaults obtained from env variables.
+pub(crate) const ENDPOINT_INVALID: &str = "https://invalid";
 
 #[derive(Derivative)]
 #[derivative(Debug)]
 #[derive(Clone, Deserialize, JsonSchema)]
 #[serde(deny_unknown_fields)]
 pub(crate) struct Config {
     /// The Apollo Studio endpoint for exporting traces and metrics.
-    #[schemars(with = "Option<String>")]
+    #[schemars(with = "Option<String>", default)]
+    #[serde(default = "endpoint_default")]
     pub(crate) endpoint: Option<Url>,
```

Comment:
If it's a default, we should not wrap it into an `Option<>`

Reply:
[3152c3c](https://github.com/apollographql/router/pull/1970/commits/3152c3c206c6f686bfd1ee12ed842c6e26b75af5)